### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by aks-**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -202,16 +200,14 @@ module.exports = function drivex(driver, wd) {
      * @returns {WebElementPromise} resolves to true or throw error
      */
     validateText: function (locator, parentWebElement, expectedText) {
-      var d = wd.promise.defer();
       methods.find(locator, parentWebElement,expectedText).getText().then(function (actual){
```
Comment (grawk) [inline index.js]: Don't we still need to return a promise out of this method?

File: index.js [hunk 2]
```diff
@@ -117,7 +117,6 @@ module.exports = function drivex(driver, wd) {
      * @returns {Promise} resolves to a WebeElement.click() (which resolves itself to a Promise
      */
     selectByOptionText: function (locator, optionText, parentWebElement) {
-      var d = wd.promise.defer();
```
Comment (grawk) [inline index.js]: In the case of selectByOptionText, we are translating the callback-based API of async.whilst into the Promise-based API of selenium. So, we do need the `defer` in this case.

File: index.js [hunk 3]
```diff
@@ -222,16 +218,14 @@ module.exports = function drivex(driver, wd) {
      * @returns {WebElementPromise} resolves to true or throw error
      */
     validateAttributeValue: function (locator, parentWebElement,attribute, expectedText) {
-      var d = wd.promise.defer();
       methods.find(locator, parentWebElement,expectedText).getAttribute(attribute).then(function (actual) {
```
Comment (grawk) [inline index.js]: need a `return` stmt to give this promise back to the caller

File: index.js (hunk 2 above)
Comment (aks-) [inline index.js]: Aha, makes sense. I just fixed it.

File: index.js (hunk 1 above)
Comment (aks-) [inline index.js]: Yes we need to return the promise :( Sorry I am not sure how I forgot to put a return in front.

**Comment by aks-** [comment]
Can you please check the changes again on this branch?

File: index.js [hunk 4]
```diff
@@ -202,16 +203,14 @@ module.exports = function drivex(driver, wd) {
...
         log('validateText : actual : ' + actual + ' expected : ' + expectedText);
         if (actual === expectedText) {
-          d.fulfill(true);
+          return true;
         } else {
-          d.reject(new Error('couldn\'t find text: ' + expectedText));
+          throw new Error('couldn\'t find text: ' + expectedText);
         }
       });
-      return d;
```
Comment (nikulkarni) [inline index.js:214]: We still need to return promises, we cannot return plain true/false to be consistent with webdriver api. Can you revert this please

File: index.js [hunk 5]
```diff
@@ -222,16 +221,14 @@ module.exports = function drivex(driver, wd) {
...
         log('validateAttributeValue : actual : ' + actual + ' expected : ' + expectedText);
         if (actual === expectedText) {
-          d.fulfill(true);
+          return true;
         } else {
-          d.reject(new Error('couldn\'t find text: ' + expectedText));
+          throw new Error('couldn\'t find text: ' + expectedText);
         }
       });
-      return d;
```
Comment (nikulkarni) [inline index.js:234]: Same as above comment

**Comment by aks-** [comment]
Hi! Sorry, been away, will have a look

**Comment by grawk** [comment]
going to close this for now. if the original author wants to re-open (@aks-) then that's fine.

**Pull request closed without merging**